src/test/storage/iterables_test.cpp

```diff
@@ -231,7 +231,24 @@ TEST_F(IterablesTest, ReferenceSegmentIteratorWithIterators) {
 
   EXPECT_EQ(sum, 24'825u);
   EXPECT_EQ(accessed_offsets,
-            (std::vector<ChunkOffset>{ChunkOffset{0}, ChunkOffset{1}, ChunkOffset{2}, ChunkOffset{3}}));
+            (std::vector<ChunkOffset>{ChunkOffset{0}, ChunkOffset{1}, ChunkOffset{2}, ChunkOffset{3}, ChunkOffset{4}}));
+}
+
+TEST_F(IterablesTest, ReferenceSegmentIteratorWithIteratorsSingleChunk) {
+  auto pos_list = PosList{NULL_ROW_ID};
+  pos_list.guarantee_single_chunk();
+
+  auto reference_segment =
+      std::make_unique<ReferenceSegment>(table, ColumnID{0u}, std::make_shared<PosList>(std::move(pos_list)));
+
+  auto iterable = ReferenceSegmentIterable<int>{*reference_segment};
+
+  auto sum = uint32_t{0};
+  auto accessed_offsets = std::vector<ChunkOffset>{};
+  iterable.with_iterators(SumUpWithIterator{sum, accessed_offsets});
+
+  EXPECT_EQ(sum, 0u);
+  EXPECT_EQ(accessed_offsets, (std::vector<ChunkOffset>{ChunkOffset{0}}));
```

Comment:
Mmh, I am not sure how safe this test is. A sum ending up to be zero can also happen if we read zero-initialized memory. Can't we explicitly check for NULL?

Reply:
We check in line 251 if the row containing NULL was accessed. And the test is not about the sum, I don't know whether a change is necessary here.